crates/oxc_traverse/src/context/scoping.rs

```diff
@@ -382,6 +382,30 @@ impl TraverseScoping {
         let symbol_id = reference.symbol_id();
         self.create_reference_id(ident.span, ident.name.clone(), symbol_id, flags)
     }
+
+    /// Determine whether evaluating the specific input `node` is a consequenceless reference.
+    ///
+    /// I.E evaluating it won't result in potentially arbitrary code from being ran. The following are
+    /// allowed and determined not to cause side effects:
+    ///
+    /// - `this` expressions
+    /// - `super` expressions
+    /// - Bound identifiers
+    ///
+    /// Based on Babel's `scope.isStatic` logic.
+    /// <https://github.com/babel/babel/blob/419644f27c5c59deb19e71aaabd417a3bc5483ca/packages/babel-traverse/src/scope/index.ts#L557>
+    #[inline]
+    pub fn is_static(&self, expr: &Expression) -> bool {
+        match expr {
+            Expression::ThisExpression(_) | Expression::Super(_) => true,
+            Expression::Identifier(ident) => ident.reference_id.get().is_some_and(|reference_id| {
+                self.symbols.get_reference(reference_id).symbol_id().is_some_and(|symbol_id| {
+                    self.symbols.get_resolved_references(symbol_id).all(|r| !r.is_write())
+                })
+            }),
```

Comment:
I think it'd be better to `unwrap` on `ident.reference_id.get()`. In the transformer, semantic has been run, so all `IdentifierReference`s should have a `ReferenceId`. Using `is_some_and` suggests that it's optional, when it's not.

Reply:
Great suggestion